source/line.cpp

```diff
@@ -14,11 +14,11 @@ struct Line{
     Line(Point<ll>& p, ld m){
         a = m; b = 1.0; c = -((a * p.x) + (b * p.y));}
     bool check_parallel(Line l){
-        return (fabs(a-l.a) < EPS) && (fabs(b-l.b) < EPS);}
+        return (abs(a-l.a) < EPS) && (abs(b-l.b) < EPS);}
     bool check_same(Line l) {                
-        return this -> check_parallel(l) && (fabs(c-l.c) < EPS);}
+        return this -> check_parallel(l) && (abs(c-l.c) < EPS);}
     bool check_orthogonal(Line l){
-        ld diff = fabs(a) - fabs(1/l.a); return (diff < EPS);}
+        ld diff = abs(a) - abs(1/l.a); return (diff < EPS);}
```

Comment:
Isn't this wrong? two lines with a=3 and a=1/3 would be considered orthogonal. This should probably be `return abs(a + 1/l.a) < EPS`

Reply:
you're right, it's wrong. It should be a, -1/a for orthogonality.

Reply:
yup, you're right. thanks, fixed👍🏻